Fix #5312: Fixes Favicon fetching to fetch from Brave-Core

Brandon-T opened the pull request:
## Summary of Changes
- Adds favicon fetching from Brave-Core
- Adds favicon fetching for Sync, History, Bookmarks, Passwords, Widgets, Tabs, NTP, Favourites from Brave-Core
- Does not migrate current favicons to Brave-Core

<!-- Enter a ticket number for this PR, create a new one if it is not there yet. -->
This pull request fixes #5312

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`


## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

soner-yuksel reviewed (commented):
Just couple of questions. thanks for this conversion and UIImageView integration for it.
I will be updating core 1.41 version in a different PR very soon and we can rebase.

Brandon-T committed:
Cache icons for domain url

Brandon-T committed:
Add SVG Support for icons

Brandon-T committed:
Favicon Rewrite
Update icon renderer
Moving everything to BraveShared
Adding separate renderers for custom and bundled icons vs. fetched icons
Removing FaviconMO
Porting all APIs that use FaviconMO to use the new APIs
Porting all APIs that use Favicons to use the new APIs
Cleaning up API

Bump brave-core to 1.41.82

kylehickinson reviewed (changes requested):
1. Lets move everything favicon related to its own `Favicons` module instead of shoving it in BraveShared and Client
2. Why did we choose to use `DispatchWorkItem` everywhere instead of Swift concurrency? 
3. Has this been tested with the favourites widget? I notice we now have 2 types of favicons, but not sure how this affects that
4. Has this been tested with multiple font sizes

Brandon-T committed:
Make method names the same

Brandon-T committed:
Move script to `.defaultClient` content world.
Allow rendering non-transparent images.
